### PR TITLE
DataCollection: introduce LOM

### PR DESCRIPTION
**Pull request by schmitz-ilias**
This PR introduces LOM for `DataCollection` as described in [Learning Object Metadata for Data Collection](https://docu.ilias.de/go/wiki/wpage_8114_1357). For details on what this entails, see the [documentation](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/components/ILIAS/MetaData/docs/enabling_lom.md).

It also adds an extremely bare-bones PRIVACY.md. If you prefer to add one properly at some point in the future let me know, then I'll remove it from this PR.

Cheers, @schmitz-ilias

### CODE REVIEW

File: components/ILIAS/DataCollection/classes/class.ilDataCollectionExporter.php
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/class.ilDataCollectionExporter.php]:
```suggestion
        if ($md_ids !== []) {
```

**Review by iszmais — APPROVED** [review]
Thank you a lot for your contribution!
LGTM, I just have to minor change requests:
- Resolve the opened thread for a bit more type strictness (we want to prevent mixed value functions if possible).
- Currently, there is no LOM Subtab implemented as it is described within the FR. Pls add this implementation or update the FR (Not the concept itself but add a note in implementation where you state why you did not implement it). I'm fine with both solutions.

Greetings,
@iszmais

**Comment by oliversamoila** [comment]
Thank you @schmitz-ilias
In general, I agree that the DataCollection should support metadata implementation to the same level as other objects. Thanks for that.
Nevertheless, as we have already discussed, I believe that we should think about the central deactivation of the service and such options for the next release. It's yet another complex function block on this object - although I think the number of people who absolutely need it is very limited. Let us talk about this for ILIAS 11.

I would like to ask you to remove the PRIVACY.md part.
The content regarding LOM is certainly correct. However, nothing else is ‘work in progress’ at the moment and so I don't want to give that impression either.

Greetings and many thanks again
@oliversamoila

**Comment by schmitz-ilias** [comment]
Hi all,

thanks for the quick response! I've made the suggested changes.

@iszmais, the subtab is handled by `ilObjectMetaDataGUI::setSubTabs`, no need to implement it in consuming components. That being said, for Data Collections it is only actually displayed when the object is in a Category with Taxonomies enabled (otherwise it is the only subtab of 'Metadata', and thus not shown). I'll clarify this in the FW along with the rest of 'Implementation' as soon as this PR is merged.

@oliversamoila, I've removed the PRIVACY.md. Regarding disabling LOM for objects, I'm sure we can work something out for ILIAS 11. There are some conceptual questions that need to be answered first (what exactly should be disabled?), I'm open to discuss options.

Cheers, @schmitz-ilias

**Comment by iszmais** [comment]
@pascalseeland Due to the reach of this change I would like you to have another look add this and would appreciate your additional approval.

**Comment by schmitz-ilias** [comment]
Hi @pascalseeland, @iszmais, @oliversamoila,

just a quick reminder that this PR still requires your collective approval. Let me know if there is anything else I can do to hurry this along.

(By the way, as far as I can tell the failing checks are due to an unrelated issue in the trunk.)

Cheers, Tim

**Comment by iszmais** [comment]
@schmitz-ilias Thank you for your contribution!

Since it's already approved by myself and no one stated something against it I see no issue in merging this.

Greetings,
@iszmais

**Comment by schmitz-ilias** [comment]
Thanks a lot @iszmais !

**Pull request merged**
